GCS_MAVLink: add MAV_SEV_LVL param

tatsuy opened the pull request:
This PR allows a user select the MAV_SEVERITY level and notifies the GCS only of the statustext that is equal to or higher than the set value.
Since the default is to send all messages, there is no impact unless the user changes the setting.

There are many functions that return false without any outputs.
In the future I want to increase messages of MAV_SEVERITY_DEBUG and make it not send by default and enable it when investigating something.

I tested this in SITL with each vehicle.

peterbarker commented:
It is an interesting idea, and I do already have a use for the sending very low priority messages.

There are a couple of downsides:
 - over-use will also cause dataflash logging contention
 - people might turn the messages all the way down, and a lot of the time these messages really, really help us debug problems when all we have is a tlog.
 - shouldn't this be a per-link thing?  There's nowhere to hang that ATM,. but we did recently merge the PRIVATE flag from @WickedShell which should really have been in a similar set of parameters.

tatsuy commented:
Thank you for the review.

- shouldn't this be a per-link thing?
I have already created another version that can be set for each channel. The param name is SRx_SEV_LVL.
https://github.com/ArduPilot/ardupilot/compare/master...tatsuy:pr-srx-sev-lvl-param?expand=1
But, I can also use SERIALx_ as same as private flag.

I have noticed if changing the param of this PR to hide the message, it is also not written to dataflash log.
The above method will solve that too.

tridge commented:
needs to be re-worked to be in AP_Vehicle, and also to still log the msgs to AP_Logger, so only avoids the mavlink msg

rmackay9 commented:
Let's name the parameter, "MAV_TEXT_SEV"

tatsuy commented:
@tridge , @rmackay9 Thank you for the review.

> needs to be re-worked to be in AP_Vehicle, and also to still log the msgs to AP_Logger, so only avoids the mavlink msg

I fixed this.
and changed the parameter name to "MAV_TEXT_SEV".

I re-tested in SITL with each vehicle.

tatsuy commented:
Added `@Values` and rebased.
